core/src/main/scala/org/broadinstitute/dsde/rawls/model/SamModel.scala

```diff
@@ -39,6 +40,12 @@ object SamProjectRoles {
   val owner = SamResourceRole("owner")
 }
 
+object SamWorkflowCollectionRoles {
+  val owner = SamResourceRole("owner")
```

Comment:
does it make sense to make this an ADT?
something like
```
sealed abstract class SamWorkflowCollectionRole{
  def role: SamResourceRole
}
object SamWorkflowCollectionRoles{
  case object Owner extends SamWorkflowCollectionRole{
    val role: SamResourceRole = SamResourceRole("owner")
  }  
  case object Writer extends SamWorkflowCollectionRole{
    val role: SamResourceRole = SamResourceRole("writer")
  }  
  case object Reader extends SamWorkflowCollectionRole{
    val role: SamResourceRole = SamResourceRole("reader")
  }
}
```
Would've been nice if `SamResourceRole` is also an ADT

Reply:
@dvoet do you have an opinion here? i think this stuff is in libs too

Reply:
I look at these things as the equivalent of string constants, we have them mostly so we don't have the same string all over the place that is then hard to change. We never actually pattern match on them or do other things where an ADT would be useful. There are plenty of places where we swap around roles and policy names on purpose and an ADT would just get in the way. Seems like work for little or no gain.

But if it were an ADT, I would be fine with it.